src/plugins/data/public/plugin.ts

```diff
@@ -91,23 +128,33 @@ export class DataPublicPlugin implements Plugin<DataPublicPluginSetup, DataPubli
 
   public start(core: CoreStart, { uiActions }: DataStartDependencies): DataPublicPluginStart {
     const { uiSettings, http, notifications, savedObjects, overlays } = core;
-    const fieldFormats = this.fieldFormatsService.start();
     setNotifications(notifications);
-    setFieldFormats(fieldFormats);
     setOverlays(overlays);
-    setUiSettings(core.uiSettings);
+    setUiSettings(uiSettings);
+
+    const fieldFormats = this.fieldFormatsService.start();
+    setFieldFormats(fieldFormats);
+
+    const indexPatterns = new IndexPatternsService(uiSettings, savedObjects.client, http);
+    setIndexPatterns(indexPatterns);
 
-    const indexPatternsService = new IndexPatternsService(uiSettings, savedObjects.client, http);
-    setIndexPatterns(indexPatternsService);
+    const query = this.queryService.start(savedObjects);
+    setQueryService(query);
+
+    const search = this.searchService.start(core);
+    setSearchService(search);
 
     uiActions.attachAction(APPLY_FILTER_TRIGGER, uiActions.getAction(ACTION_GLOBAL_APPLY_FILTER));
 
     const dataServices = {
+      actions: {
+        createFiltersFromEvent,
+      },
```

Comment:
`createFiltersFromEvent` is technically stateful as it relies on index patterns behind the scenes. Open to suggestions on where this should live, but in its current form it cannot be exported statically @ppisljar